### DISCUSSION TITLE
GUID purpose on Each driver

### DISCUSSION THREAD

**Discussion by dinguluer** [grodansparadis/vscp#272] [Q&A] [answered]
Hi,

What is purpose of having GUID in each driver ?

If we consider the case of RAW ethernet driver ... end node will have its own GUID.
Then will be the use of GUID for  RAW ethernet driver ?

![image](https://github.com/grodansparadis/vscp/assets/370819/3f18878c-7558-4c52-97a6-86e1505cc430)

Regards,
Dinesh

**Comment by grodansparadis** [comment, marked as answer]
Some networks does (like CAN, LoRa etc) use nicknames instead of GUID's in there communication. A node like that still has a real GUID but don't use that GUID when it communicates with the world. Instead it use the drivers GUID with it's own nickname in the two lsb bytes of the GUID, The driver thus acts as a proxy for the node. It is still possible to read the GUID from the node and it is still identified by it.

Driver GUID's always has the two lsb bytes set to zero.
